bool training_fg
bool replay_fg
bool training
bool replay
